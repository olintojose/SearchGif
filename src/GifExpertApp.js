import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrids } from './components/GifGrids';

export const GifExpertApp = ({ defaultCategories = [] }) => {
const [categories, setCategories] = useState( defaultCategories );


    return ( 
        <>
       
        <h2 className="animate__animated animate__flipInX">Search Gif</h2>
        <AddCategory setCategories={ setCategories }/>
        <hr/>
        
        <ol>
            {
                categories.map(category => 
                  <GifGrids 
                   key={category}
                   category={category} />
                )
            }
        </ol>
        </>
     );
}
