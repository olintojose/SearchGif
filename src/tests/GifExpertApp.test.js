import React from 'react'

import { shallow } from 'enzyme'
import { GifExpertApp } from '../GifExpertApp'


describe('Pruebas en GifExpertApp', () => {
    test('Debe mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect (wrapper).toMatchSnapshot();

    })

    test('Debe mostrar la lista de categorias', () => {
        
        const categories = ['Goku', 'Messi'];

        const wrapper = shallow(<GifExpertApp defaultCategories= {categories} />);

       // console.log(categories.length)

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrids').length).toBe(categories.length);
    })
    
    
})
