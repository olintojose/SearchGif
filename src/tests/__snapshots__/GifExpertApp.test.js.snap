// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe mostrar la lista de categorias 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__flipInX"
  >
    Search Gif
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrids
      category="Goku"
      key="Goku"
    />
    <GifGrids
      category="Messi"
      key="Messi"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__flipInX"
  >
    Search Gif
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
